DEVPROD-5307: Documentation for Parsley Sectioning (#8346)
evergreen-ci/evergreen: docs/UI/Parsley-Sectioning.md

# Parsley Sectioning

## Overview

> Parsley sectioning is toggleable from the "Log Viewing" tab of the "Parsley Settings" menu and is available to task logs only. Toggling the feature will render your task log into expandable sections split up by function and command.

Reviewer: ```suggestion
Parsley sectioning is toggleable from the "Log Viewing" tab of the "Parsley Settings" menu and is available for task logs only. Enabling sectioning will render your task log into expandable sections split up by function and command.
```

> ## Feature details

Reviewer: nit. Would just recommend double checking if you want these headers in title case or in sentence case since it's slightly inconsistent (e.g. `Section status` vs `Jump to Failing Line`)

Author: Good one. I'll stick to sentence case.

### Feature components
At the top of the log viewer, two buttons are available to "Open all sections" and "Close all sections". Clicking these buttons will expand and collapse every function and command section in the log file.

![Toggle All Sections](../images/toggle_all_sections.png)

> There are also toggles available on function rows, "Open Subsections" and "Close Subsections", to expand and collapse all commands that belong to the function.

Reviewer: would recommend changing `toggle` => `button`
```suggestion
Similarly there are "Open Subsections" and "Close Subsections" buttons available on individual function rows, which can be used to expand and collapse all commands that belong to the function.
```

![Toggle Subsections](../images/toggle_subsections.png)

> Clicking on the caret will also open and close a single command or function section.
> 
> ![Toggle Single Section](../images/section_toggle.png)

Reviewer: I think you can omit this image (since it looks kinda odd just having the image of the caret) and just say
```suggestion
The caret on the left-hand side can also be used to open or close a single command or function section.
```
which the reader can infer from the picture directly above

### Section status
A function containing a failing command is marked with a failing status icon whereas as passing function is marked with a passing icon.

![Passing Function](../images/passing_function.png)

![Failing Function](../images/failing_function.png)

### Jump to Failing Line
Jump to Failing Line is a separate feature that works synergistically with Sectioning. When enabled with Sectioning, Parsley will open the failing command section and scroll to the failing line during initial load.

### Searching

> Sections that contain a search result will open. Iterating over search results will open the section if it's closed.

Reviewer: ```suggestion
Sections that match the search criteria will open automatically.
```

### Filtering

> Sections will temporarily go away when filters are applied and come back when filters are removed.

Reviewer: (or something along these lines)
```suggestion
Sectioning cannot be used in conjunction with filters. Sections will not render when filters are applied, but will reappear when all filters are removed.
```

> ### Shareline
> When a link with a share line is opened, Parsley will open the section that contains the share line and scroll to the line. The failing line section will still open but Parsley will scroll to the share line instead.

Reviewer: ```suggestion
### Share Line
Share line takes precedence over failing line. When a link with a share line is opened, Parsley will open the section that contains the share line and scroll to the line. The failing line section will still be open and accessible via the bookmarks bar.
```
